service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java

```diff
@@ -177,6 +177,38 @@ void numberDirections_shouldApplyCorrectNumberingWhenDirectionsAreMarkedAsNotNee
         assertThat(numberedDirectionTypes).isEqualTo(expectedDirectionTypes);
     }
 
+    @Test
+    void removeUnnecessaryDirections_shouldRemoveDirectionsWhenDirectionsAreMarkedAsNotNeeded() {
+        List<Direction> directions = unwrapElements(service.removeUnnecessaryDirections(directionsMarkedAsRemoved()));
+
+        List<Direction> expectedDirections = unwrapElements(Stream.of(buildDirections(ALL_PARTIES, "Yes"),
+            buildDirections(CAFCASS, "Yes"),
+            buildDirections(COURT, "Yes"))
+            .flatMap(Collection::stream)
+            .collect(toList()));
+
+        assertThat(directions).isEqualTo(expectedDirections);
+    }
+
+    @Test
+    void changeSdoTabView_shouldApplyTabViewStatusToListOfDirectionsWhenParametersProvided() {
+        Direction direction = Direction.builder()
+            .directionType("direction")
+            .directionText("example direction text")
+            .directionNeeded("Yes")
+            .assignee(LOCAL_AUTHORITY)
+            .tabView(null)
+            .build();
+
+        List<Direction> directions = unwrapElements(
+            service.changeSdoTabView(newArrayList(element(direction)), "Yes"));
+
+        List<Direction> expectedDirections = unwrapElements(
+            newArrayList(element(direction.toBuilder().tabView("Yes").build())));
```

Comment:
unwrapElements on element? why not just newArrayList(direction....)

Reply:
woops :D resolved